### PR TITLE
Allow compile() from outside the plugin

### PR DESCRIPTION
**Pull request by niaccurshi**
Cleaned up the class-wp-scss.php file primarily to allow for multiple instances of the Wp_Scss class without conflict with internal variables. 

Removed almost all global variable references to create a more class based/object oriented structure, including removing direct calls to public class variables and replacing with function calls to retrieve private variables.

### CODE REVIEW

File: class/class-wp-scss.php [hunk 1]
```diff
@@ -25,27 +25,62 @@ class Wp_Scss {
...
    */
   public function __construct ($scss_dir, $css_dir, $compile_method, $sourcemaps) {
-    global $scssc;
+  
     $this->scss_dir         = $scss_dir;
     $this->css_dir          = $css_dir;
     $this->compile_method   = $compile_method;
     $this->compile_errors   = array();
-    $scssc                  = new Compiler();
+    $this->scssc                  = new Compiler();
```
Comment (shadoath) [inline class/class-wp-scss.php:33]: Please match the indent of the `=` sign with the line above for better code readablity.

File: class/class-wp-scss.php [hunk 2]
```diff
@@ -54,91 +89,93 @@ public function __construct ($scss_dir, $css_dir, $compile_method, $sourcemaps)
...
-    //Compiler - Takes scss $in and writes compiled css to $out file
-    // catches errors and puts them the object's compiled_errors property
-    if (!function_exists( 'compiler' )) {
-      function compiler($in, $out, $instance) {
-        global $scssc, $cache;
+    // For each input file, find matching css file and compile
+    foreach ($input_files as $scss_file) {
+      $input = $this->scss_dir.$scss_file;
+      $outputName = preg_replace("/\.[^$]*/",".css", $scss_file);
+      $output = $this->css_dir.$outputName;
```
Comment (shadoath) [inline class/class-wp-scss.php:106]: Add spaces around the `.` for better readability and to maintain code style.
```suggestion
      $input = $this->scss_dir . $scss_file;
      $outputName = preg_replace("/\.[^$]*/", ".css", $scss_file);
      $output = $this->css_dir . $outputName;
```

File: class/class-wp-scss.php [hunk 3]
```diff
@@ -54,91 +89,93 @@ public function __construct ($scss_dir, $css_dir, $compile_method, $sourcemaps)
...
-            array_push($instance->compile_errors, $errors);
+      $this->compiler($input, $output, $this);
+    }
+
+    if (count($this->compile_errors) < 1) {
+      if  ( is_writable($this->css_dir) ) {
+        foreach (new DirectoryIterator($this->cache) as $this->cache_file) {
+          if ( pathinfo($this->cache_file->getFilename(), PATHINFO_EXTENSION) == 'css') {
+            file_put_contents($this->css_dir.$this->cache_file, file_get_contents($this->cache.$this->cache_file));
+            unlink($this->cache.$this->cache_file->getFilename()); // Delete file on successful write
```
Comment (shadoath) [inline class/class-wp-scss.php:116]:
```suggestion
            file_put_contents($this->css_dir . $this->cache_file, file_get_contents($this->cache . $this->cache_file));
            unlink($this->cache . $this->cache_file->getFilename()); // Delete file on successful write
```

File: class/class-wp-scss.php [hunk 4]
```diff
@@ -54,91 +89,93 @@ public function __construct ($scss_dir, $css_dir, $compile_method, $sourcemaps)
...
-            }
-          }
-        } else {
-          $errors = array(
-            'file' => 'CSS Directory',
-            'message' => "File Permissions Error, permission denied. Please make your CSS directory writable."
-          );
-          array_push($this->compile_errors, $errors);
-        }
+        file_put_contents($this->cache.basename($out), $css);
```
Comment (shadoath) [inline class/class-wp-scss.php:162]:
```suggestion
        file_put_contents($this->cache . basename($out), $css);
```

File: wp-scss.php [hunk 5]
```diff
@@ -266,13 +266,15 @@ function wpscss_settings_show_errors($errors) {
...
     // Show in the header to anyone
-  } else if ( !is_admin() && $wpscss_settings['errors'] === 'show' && count($wpscss_compiler->compile_errors) > 0) {
-    wpscss_settings_show_errors($wpscss_compiler->compile_errors);
+  } else if ( !is_admin() && $wpscss_settings['errors'] === 'show' && count($compile_errors) > 0) {
+    wpscss_settings_show_errors($compile_errors);
   } else { // Hide errors and print them to a log file.
-    foreach ($wpscss_compiler->compile_errors as $error) {
+    foreach ($compile_errors as $error) {
       $error_string = date('m/d/y g:i:s', time()) .': ';
       $error_string .= $error['file'] .' - '. $error['message'] . PHP_EOL;
```
Comment (shadoath) [inline wp-scss.php:279]:
```suggestion
      $error_string .= $error['file'] . ' - ' . $error['message'] . PHP_EOL;
```

**Review by shadoath — APPROVED** [review]
Really good work on this. Thank you for the clean-up as well as the added functionality. I added some minor code styling requests (I think these may have been around before your changes but I'm only seeing them now).

**Comment by niaccurshi** [comment]
Thanks, some of them may well have been me, either way I've committed those changes.

**Comment by shadoath** [comment]
Resolves #182

**Pull request merged**